src/include/api.h

```diff
@@ -33,49 +33,58 @@
 #endif
 
 #define API_SESSION_PUSH(s, h, n, dh)       \
+    WT_CURSOR *__cursor = NULL;             \
     WT_DATA_HANDLE *__olddh = (s)->dhandle; \
     const char *__oldname = (s)->name;      \
+    bool __op_timer_started = false;        \
```

Comment:
We should record whether we have started the timer and then only stop it if we have started it. With this, no special handling for the close case is required.

Reply:
Very nice!